logic/common/utils_file.py

```diff
@@ -0,0 +1,191 @@
+'''
+This file contains:
+ - Custom printing funcitons
+ - Reader function for choose the root from TOML
+
+'''
+
+#==================================================#
+'''Import & Default Constants'''
+
+import time
+import os
+import toml
+
+DEFAULT_ROOT_TOML = "input/root_dir.toml"
+
+
+
+#==================================================#
+'''Print Functions'''
+
+def print_linebreak(upper_spacing = 0, lower_spacing = 0, linebreak_char = '='):
+	'''Prints a divider, can add extra spacing above and/or below'''
+	for spacing_count in range(upper_spacing):
+		print("")
+	print(linebreak_char*50)
+	for spacing_count in range(lower_spacing):
+		print("")
+
+def _print(print_str):
+	'''Alternate print function'''
+	print(f"\t{print_str}")
+
+def _print_list(list_s):
+	'''Print all the contents of a list of string'''
+	for line in list_s:
+		print(line)
+
+
+
+#==================================================#
+'''File I/O'''
+
+def check_file_exist(filepath):
+	'''Return False if file is not found, may remove since it's just one-liner'''
+	return os.path.exists(filepath)
+
+def get_first_valid_root(root_dir_toml):
+	'''Return filepath based on the users' computer'''
+	root_dir_args = toml.load(root_dir_toml)
+	for name, root in root_dir_args.items():
+		if os.path.isdir(root):
+			return root + "/"
+	return NONE
+
+
+
+#==================================================#
+'''Untitided'''
+
+
+
+
+
+# DO NOT READ
+# WILL TIDY, SALVAGE, OR REMOVE WHEN NEEDED
```

Comment:
Noo! Please don't do this. Bad, bad, bad code etiquette. To give you an analogy, it's like if your mom was going out to get some groceries, and she said, "do you need anything"?

And then you asked her to get you a PS5 O_o

Your mom is going out for groceries - not an electronics store. Also, a PS5 is a big ask.

The whole point of branches is that they are cheap and disposable - and there are a lot of them. We can juggle multiple branches at once too. Each branc/pull request addresses one thing. This particular branch was already a bit bloated - don't suddenly introduce other content in the middle of it.

Please get rid of it

Reply:
Not too sure about the PS5 analogy, but I get your point. Removed now.